x-pack/plugins/lens/public/visualizations/xy/xy_config_panel/annotations_config_panel/range_annotation_panel.tsx

```diff
@@ -0,0 +1,154 @@
+            const newPointAnnotation: PointInTimeEventAnnotationConfig = {
+              type: 'manual',
+              key: {
+                type: 'point_in_time',
+                timestamp: annotation.key.timestamp,
+              },
+              id: annotation.id,
+              label:
+                annotation.label === defaultRangeAnnotationLabel
+                  ? defaultAnnotationLabel
+                  : annotation.label,
+              color: toLineAnnotationColor(annotation.color),
+              isHidden: annotation.isHidden,
+            };
+            onChange(newPointAnnotation);
+          } else if (annotation) {
+            const fromTimestamp = moment(annotation?.key.timestamp);
+            const dataLayers = getDataLayers(state.layers);
+            const newRangeAnnotation: RangeEventAnnotationConfig = {
+              type: 'manual',
+              key: {
+                type: 'range',
+                timestamp: annotation.key.timestamp,
+                endTimestamp: getEndTimestamp(
+                  datatableUtilities,
+                  fromTimestamp.toISOString(),
+                  frame,
+                  dataLayers
+                ),
+              },
+              id: annotation.id,
+              label:
+                annotation.label === defaultAnnotationLabel
+                  ? defaultRangeAnnotationLabel
+                  : annotation.label,
+              color: toRangeAnnotationColor(annotation.color),
+              isHidden: annotation.isHidden,
+            };
+            onChange(newRangeAnnotation);
+          }
+        }}
+        compressed
+      />
+    </EuiFormRow>
+  );
+};
+
+export const ConfigPanelRangeDatePicker = ({
+  value,
+  label,
+  prependLabel,
+  onChange,
+  dataTestSubj = 'lnsXY_annotation_date_picker',
+}: {
+  value: moment.Moment;
+  prependLabel?: string;
+  label?: string;
+  onChange: (val: moment.Moment | null) => void;
+  dataTestSubj?: string;
+}) => {
+  return (
+    <EuiFormRow display="rowCompressed" fullWidth label={label} className="lnsRowCompressedMargin">
+      {prependLabel ? (
+        <EuiFormControlLayout
+          fullWidth
+          className="lnsConfigPanelNoPadding"
+          prepend={
+            <EuiFormLabel className="lnsConfigPanelDate__label">{prependLabel}</EuiFormLabel>
+          }
+        >
+          <EuiDatePicker
+            calendarClassName={DONT_CLOSE_DIMENSION_CONTAINER_ON_CLICK_CLASS}
+            fullWidth
+            showTimeSelect
+            selected={value}
+            onChange={onChange}
+            dateFormat="MMM D, YYYY @ HH:mm:ss.SSS"
+            data-test-subj={dataTestSubj}
+          />
+        </EuiFormControlLayout>
+      ) : (
+        <EuiDatePicker
+          calendarClassName={DONT_CLOSE_DIMENSION_CONTAINER_ON_CLICK_CLASS}
+          fullWidth
+          showTimeSelect
+          selected={value}
+          onChange={onChange}
+          dateFormat="MMM D, YYYY @ HH:mm:ss.SSS"
```

Comment:
It would be nice to use the `dateFormat` advanced setting here instead of hardcoding a format

Reply:
Opened a follow-up issue for that